.github/workflows/prerelease_macos.yml

```diff
@@ -0,0 +1,150 @@
+  GO_VERSION: '1.14.4'
+  GPG_MAIL: 'infrastructure-eng@newrelic.com'
+  GPG_PASSPHRASE: ${{ secrets.OHAI_GPG_PASSPHRASE }}
+  GPG_PRIVATE_KEY_BASE64: ${{ secrets.OHAI_GPG_PRIVATE_KEY_BASE64 }} # base64 encoded
+  SNYK_TOKEN: ${{ secrets.CAOS_SNYK_TOKEN }}
+  TAG:  ${{ github.event.release.tag_name }}
+  DOCKER_HUB_ID: ${{ secrets.OHAI_DOCKER_HUB_ID }}
+  DOCKER_HUB_PASSWORD: ${{ secrets.OHAI_DOCKER_HUB_PASSWORD }}
+  SCHEMA_BRANCH: ${{ github.event.release.tag_name }}
+  AWS_S3_BUCKET_NAME: "nr-downloads-ohai-staging"
+  AWS_S3_LOCK_BUCKET_NAME: "onhost-ci-lock-staging"
+  AWS_REGION: "us-east-1"
+  DOCKER_PUBLISH: true
+
+jobs:
+  unit-test-macos:
+    name: macOS unit tests
+    runs-on: macos-11
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{env.GO_VERSION}}
+
+      - name: Validate code
+        run: make ci/validate
+
+      - name: Running unit tests
+        run: make ci/test-coverage
+
+      - name: Coveralls Parallel
+        uses: coverallsapp/github-action@master
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          path-to-lcov: lcov.info
+          flag-name: run-linux
+          parallel: true
+
+  harvest-test-macos:
+    name: macOS harvest tests
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ macos-11, macos-10.15 ]
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Running Harvest tests
+        run: make macos/harvest-tests
+
+  proxy-tests-macos:
+    name: Proxy tests
+    runs-on: macos-11
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ env.DOCKER_HUB_ID }}
+          password: ${{ env.DOCKER_HUB_PASSWORD }}
+
+      - name: Running proxy tests
+        run: make proxy-test
+
+  finish:
+    name: Finish
+    needs: [ unit-test-macos ]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Coveralls Finished
+        uses: coverallsapp/github-action@master
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          parallel-finished: true
+
+  packaging-macos:
+    name: Build and upload all artifacts into GH Release assets
+    runs-on: macos-11
+    needs: [unit-test-macos, harvest-test-macos, proxy-tests-macos]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ env.DOCKER_HUB_ID }}
+          password: ${{ env.DOCKER_HUB_PASSWORD }}
+
+      - name: Releasing macos packages
+        run: make ci/prerelease/macos
+
+  publishing-to-s3-macos:
+    name: Publish macos artifacts into s3 staging bucket
+    runs-on: ubuntu-20.04
+    needs: [packaging-macos]
+
+    strategy:
+      max-parallel: 1
+      matrix:
+        assetsType:
+          - "targz"
```

Comment:
we will need to create new type & file for `macos` and put all assets there. I current implementation we will re-upload all targz again that is not nice.

Reply:
This approach splitting the whole release process for macOS is already extracting building tarballs only for macOS in one workflow and Linux ones in a different one. Therefore there'll be no clash.

Reply:
See https://github.com/newrelic/infrastructure-agent/pull/595/files#diff-cebc817ce00a1cea59f97de909e59ae74c0d447891f779d1ad109a7d23ee8f0eL50-L52